### PR TITLE
docs: add runnable code samples for DataFrames I/O methods and property

### PR DESCRIPTION
**Pull request by ashleyxuu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 1]
```diff
@@ -33,7 +33,18 @@ class DataFrame(NDFrame):
...
+        Return a tuple representing the dimensionality of the DataFrame.
+
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            >>> df = bpd.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
+            >>> df.shape
+            (2, 2)
```
[the remark this answers is not in the document]
Comment (ashleyxuu) [reply third_party/bigframes_vendored/pandas/core/frame.py:46]: Done.

**Review by tswast — APPROVED** [review]

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 2]
```diff
@@ -171,6 +221,15 @@ def to_dict(
         The type of the key-value pairs can be customized with the parameters
         (see below).
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            >>> df = bpd.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
```
Comment (shobsi) [inline third_party/bigframes_vendored/pandas/core/frame.py:229]: Can we add one example with user provided column names and show that the reflect after to_dict()?

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 3]
```diff
@@ -298,6 +386,15 @@ def to_string(
...
         """Render a DataFrame to a console-friendly tabular output.
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            >>> df = bpd.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
+            >>> df.to_string()
+            '   col1  col2\\n0     1     3\\n1     2     4'
```
Comment (shobsi) [inline third_party/bigframes_vendored/pandas/core/frame.py:396]: Would it not work if we kept the expected output multiline, encapsulated in """?

File: third_party/bigframes_vendored/pandas/core/frame.py [hunk 4]
```diff
@@ -363,6 +460,15 @@ def to_markdown(
...
         """Print DataFrame in Markdown-friendly format.
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+            >>> df = bpd.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
+            >>> df.to_string()
+            '   col1  col2\\n0     1     3\\n1     2     4'
```
Comment (shobsi) [inline third_party/bigframes_vendored/pandas/core/frame.py:470]: Would it render multiline in the generated doc?

Comment (ashleyxuu) [reply third_party/bigframes_vendored/pandas/core/frame.py:470]: AFAIU, doctest test the plain outcome in the cell. When you actually copy and paste the result in a shell, the format will render correctly. Example here:
https://screenshot.googleplex.com/9eSsyGRx43pinSu

**Review by shobsi — APPROVED** [review]

**Pull request merged**